markets/perps-market/test/integration/Market/PerpsMarketModule.test.ts

```diff
@@ -53,14 +54,88 @@ describe('PerpsMarketModule', () => {
       assertBn.equal(summary.indexPrice, fixture.marketTokenPrice);
     });
   });
+
   describe('fillPrice', () => {
-    it('should return correct value when passing same as onchain price', async () => {
-      const price = await systems().PerpsMarket.fillPrice(marketId, bn(1), bn(1000));
-      assertBn.equal(price, bn(1000.05));
+    describe('skewScale 0', () => {
+      const restoreSkewScale = snapshotCheckpoint(provider);
+      before('set skewScale to 0', async () => {
+        await systems().PerpsMarket.connect(marketOwner()).setFundingParameters(marketId, 0, 0);
+      });
+      it('should return the index price', async () => {
+        const price = await systems().PerpsMarket.fillPrice(marketId, bn(1), bn(1000));
+        assertBn.equal(price, fixture.marketTokenPrice);
+      });
+      after('restore skewScale', restoreSkewScale);
+    });
+    let commonOpenPositionProps: Pick<
+      OpenPositionData,
+      | 'systems'
+      | 'provider'
+      | 'trader'
+      | 'accountId'
+      | 'keeper'
+      | 'marketId'
+      | 'settlementStrategyId'
+    >;
+    before('identify common props', async () => {
+      commonOpenPositionProps = {
+        systems,
+        provider,
+        marketId: marketId,
+        trader: trader2(),
+        accountId: 2,
+        keeper: keeper(),
+        settlementStrategyId: bn(0),
+      };
     });
-    it('should return correct value when passing different price', async () => {
-      const price = await systems().PerpsMarket.fillPrice(marketId, bn(1), bn(1010));
-      assertBn.equal(price, bn(1010.0505));
+    const tests = [
+      {
+        marketSkew: 0,
+        sizeAndExpectedPrice: [
+          { size: 1, price: bn(1010), expectedPrice: bn(1010.0505) },
+          { size: -1, price: bn(1010), expectedPrice: bn(1009.9495) },
+        ],
+      },
+      {
+        marketSkew: 10,
+        sizeAndExpectedPrice: [
+          { size: 1, price: bn(1010), expectedPrice: bn(1011.0605) },
+          { size: -1, price: bn(1010), expectedPrice: bn(1010.9595) },
+          { size: -11, price: bn(1010), expectedPrice: bn(1010.4545) },
+        ],
+      },
+      {
+        marketSkew: -10,
+        sizeAndExpectedPrice: [
+          { size: 1, price: bn(1010), expectedPrice: bn(1009.0405) },
+          { size: -1, price: bn(1010), expectedPrice: bn(1008.9395) },
+          { size: 11, price: bn(1010), expectedPrice: bn(1009.5455) },
+        ],
+      },
+    ];
+    before('add collateral', async () => {
```

Comment:
not an issue here, but take into account that this `before` will be executed before the `describe "skewScale 0"`

Reply:
Ah yea thanks, I'll just move it to make that clear